### ISSUE TITLE
User/logs not told if Sonic Pi server not running

### ISSUE THREAD

**Issue by ird28** [cambridge-alpha-team/unvisual-backend#9]
If the Sonic Pi server isn't running, then the OSC messages are still successfully sent, so nothing is reported in our end to have gone wrong. The net effect of this is that if the Sonic Pi server is not running (or not responding or whatever), and the user tries to run their code, there will just be no response from the system at all.

It would be good to be able to let them know why it wasn't working.

**Comment by KoviRobi** [comment]
Should the OSC library tell you that it was unable to establish a connection?

**Comment by KoviRobi** [comment]
I think the problem is also in Server.java:20, sender is never initialized as far as I could see.

**Comment by ird28** [comment]
No, I don't think so. As far as I can tell (I might be wrong), when you send an OSC message, no connection is established, the message is just fired out into the wilderness.

If you want to listen for OSC messages, though, it does seem to establish a connection. This is a problem because only one listener per port seems to be allowed, and if the Sonic Pi GUI is running then it listens to the Sonic Pi server, so that we can't.

This means that it doesn't look like we'll be able to listen for the responses that the server sends when you send it messages.

**Comment by KoviRobi** [comment]
As for the sonic pi gui running, we could just run only the sonic pi server.

As for the lack of exceptions, that is because OSC uses UDP (https://github.com/hoijui/JavaOSC/blob/master/modules/core/src/main/java/com/illposed/osc/OSCPortOut.java) so not sure how we could solve that

**Comment by ird28** [comment]
Yes, that does seem to be the only solution. I'm not sure how to start the Sonic Pi server by itself, but I suppose we can look into it.

Assuming we can get that working, perhaps we could report an error if the Sonic Pi server doesn't respond for a certain amount of time?

**Comment by KoviRobi** [comment]
Looking at app/gui/qt/mainwindow.cpp it starts server/bin/sonic-pi-server.rb but starting that manually does not yet work with our backend, seems like there is some handshaking going on before we can start sending code, as it works with starting that, then the Qt app.

Not sure if implementing the whole handshaking is worth our effort given that it works if we have the gui running in the background.

**Comment by ird28** [comment]
I agree. Is there anything we can do about this issue then?

**Comment by KoviRobi** [comment]
Don't think so, hence I added the wontfix label, you can close it if you wish.
